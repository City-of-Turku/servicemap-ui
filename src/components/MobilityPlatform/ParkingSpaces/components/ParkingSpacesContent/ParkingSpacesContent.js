import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';

const ParkingSpacesContent = ({
  classes, intl, parkingSpace, parkingStatistics,
}) => {
  const renderText = (textType, translationId, text) => (
    <div className={textType === 'title' ? classes.title : classes.text}>
      {textType === 'title' ? (
        <Typography variant="subtitle1">
          {intl.formatMessage({
            id: translationId,
          })}
        </Typography>
      ) : (
        <Typography variant="body2">
          <strong>
            {intl.formatMessage({
              id: translationId,
            })}
            :
          </strong>
          {' '}
          {text}
        </Typography>
      )}
    </div>
  );

  const renderPaymentType = (translationId1, translationId2) => (
    <div className={classes.text}>
      <Typography variant="body2">
        <strong>
          {intl.formatMessage({
            id: translationId1,
          })}
          :
        </strong>
        {' '}
        {intl.formatMessage({
          id: translationId2,
        })}
      </Typography>
    </div>
  );

  const renderParkingCount = (capacity, parkingCount) => {
    const freeParkingSpaces = capacity - parkingCount;

    return (
      <div key={capacity} className={classes.text}>
        {freeParkingSpaces > 0 ? (
          <Typography variant="body2">
            <strong>
              {intl.formatMessage({ id: 'mobilityPlatform.content.parkingSpaces.parkingCount' })}
            </strong>
            :
            {' '}
            {freeParkingSpaces}
            {''}
            /
            {''}
            {capacity}
          </Typography>
        ) : (
          <Typography variant="body2">
            {intl.formatMessage({ id: 'mobilityPlatform.content.parkingSpaces.empty' })}
          </Typography>
        )}
      </div>
    );
  };

  return (
    <div className={classes.container}>
      {renderText('title', 'mobilityPlatform.content.parkingSpaces.title')}
      {renderPaymentType('mobilityPlatform.content.parkingSpaces.type', 'mobilityPlatform.content.parkingSpaces.paid')}
      {renderText('text', 'mobilityPlatform.content.parkingSpaces.capacity', parkingSpace.properties.capacity_estimate)}
      {parkingStatistics.filter(item => item.id === parkingSpace.id).map(parking => (
        renderParkingCount(parkingSpace.properties.capacity_estimate, parking.current_parking_count)
      ))}
    </div>
  );
};

ParkingSpacesContent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  parkingSpace: PropTypes.objectOf(PropTypes.any),
  parkingStatistics: PropTypes.arrayOf(PropTypes.any),
};

ParkingSpacesContent.defaultProps = {
  parkingSpace: null,
  parkingStatistics: [],
};

export default ParkingSpacesContent;
