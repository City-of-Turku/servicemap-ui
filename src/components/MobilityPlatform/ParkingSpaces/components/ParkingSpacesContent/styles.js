export default theme => ({
  container: {
    margin: theme.spacing(1.25),
  },
  title: {
    width: '85%',
    borderBottom: '1px solid #000000',
    marginBottom: theme.spacing(1),
    paddingBottom: theme.spacing(0.8),
  },
  text: {
    paddingBottom: theme.spacing(1),
  },
});
