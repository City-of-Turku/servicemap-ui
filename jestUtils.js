import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });