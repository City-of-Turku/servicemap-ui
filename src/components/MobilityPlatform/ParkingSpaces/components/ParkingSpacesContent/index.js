import { withStyles } from '@mui/styles';
import { injectIntl } from 'react-intl';
import ParkingSpacesContent from './ParkingSpacesContent';
import styles from './styles';

export default withStyles(styles)(injectIntl(ParkingSpacesContent));
