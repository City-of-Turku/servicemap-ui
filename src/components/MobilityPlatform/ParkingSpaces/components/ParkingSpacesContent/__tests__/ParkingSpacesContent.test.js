// Link.react.test.js
import React from 'react';
import ParkingSpacesContent from '../index';
import { getRenderWithProviders } from '../../../../../../../jestUtils';
import finnishTranslations from '../../../../../../i18n/fi';

const mockProps = {
  parkingSpace: {
    id: '123abc',
    properties: {
      capacity_estimate: 10,
    },
  },
  parkingStatistics: [
    {
      id: '123abc',
      current_parking_count: 5,
    },
  ],
};

const renderWithProviders = getRenderWithProviders({});

describe('<ParkingSpacesContent />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<ParkingSpacesContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<ParkingSpacesContent {...mockProps} />);

    const h3 = container.querySelectorAll('h3');
    const p = container.querySelectorAll('p');
    expect(h3[0].textContent).toContain(finnishTranslations['mobilityPlatform.content.parkingSpaces.title']);
    expect(p[0].textContent).toContain(
      `${finnishTranslations['mobilityPlatform.content.parkingSpaces.type']}: ${finnishTranslations['mobilityPlatform.content.parkingSpaces.paid']}`,
    );
    expect(p[1].textContent).toContain(
      `${finnishTranslations['mobilityPlatform.content.parkingSpaces.capacity']}: ${mockProps.parkingSpace.properties.capacity_estimate}`,
    );
    expect(p[2].textContent).toContain(`Vapaana olevat parkkipaikat: 5 / ${mockProps.parkingSpace.properties.capacity_estimate}`);
  });
});
