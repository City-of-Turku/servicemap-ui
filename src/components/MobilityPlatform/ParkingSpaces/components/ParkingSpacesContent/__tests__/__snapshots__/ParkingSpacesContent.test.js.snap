// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParkingSpacesContent /> should work 1`] = `
<div>
  <div
    class="injectIntl(ParkingSpacesContent)-container-1"
  >
    <div
      class="injectIntl(ParkingSpacesContent)-title-2"
    >
      <h3
        class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
      >
        Pysäköintialue
      </h3>
    </div>
    <div
      class="injectIntl(ParkingSpacesContent)-text-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Maksutyyppi
        :
         
        Maksullinen
      </p>
    </div>
    <div
      class="injectIntl(ParkingSpacesContent)-text-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Parkkipaikkojen määrä
        :
         
        10
      </p>
    </div>
    <div
      class="injectIntl(ParkingSpacesContent)-text-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Vapaana olevat parkkipaikat: 5 / 10
      </p>
    </div>
  </div>
</div>
`;
